api_parser/json_to_openapi_2_0.py

```diff
@@ -0,0 +1,638 @@
+        for key in keys[:-1]:
+            dic = dic.setdefault(key, {})
+        if keys[-1] not in dic:
+            dic[keys[-1]] = value
+        elif isinstance(value, dict) and isinstance(dic[keys[-1]], dict):
+            dic[keys[-1]].update(value)
+
+    @staticmethod
+    def nested_set(dic, keys, value):
+        """Set a value in a dictionary based on the list of keys pointing to that value."""
+        # Ensure all keys in keys list are present
+        for key in keys[:-1]:
+            dic = dic.setdefault(key, {})
+            dic[keys[-1]] = value
+
+    @staticmethod
+    def nested_get(input_dict, nested_key):
+        """Get a key based on the list of keys."""
+        internal_dict_value = input_dict
+        for k in nested_key:
+            internal_dict_value = internal_dict_value.get(k, None)
+            if internal_dict_value is None:
+                return None
+        return internal_dict_value
+
+    @staticmethod
+    def response_name_to_path_list(name):
+        """Change a response name to a list of keys (the 'path') to the item in the models dict,
+        example:
+        features.0.message --> ['features', 'items', 'properties', 'message']"""
+        name_list = name.split(".")
+        l = [name_list[0]]
+        for i, name in enumerate(name_list[1:]):
+            # If the name is a number, it represents an item in the list and we should replace it with an "items" key.
+            if name.isdigit():
+                l.append('items')
+            # Else if it is a name, this must be an object, and so we need to add a "properties" key before it.
+            else:
+                l.append("properties")
+                l.append(name)
+        return l
+
+    @staticmethod
+    def convert_form_list_to_form_data_list(form_list):
+        """Used to turn the form_list into a swagger list of parameters"""
+        fl = []
+        for field in form_list:
+            f = {}
+            f['in'] = 'formData'
+            f['name'] = field['field']
+            f.update(Converter.convert_type(field['type']))
+            f['required'] = not field['optional']
+            f['description'] = field['description']
+            fl.append(f)
+        return fl
+
+    @staticmethod
+    def convert_field_list_to_parameter_list(field_list):
+        fl = []
+        for field in field_list:
+            f = {}
+            f['name'] = field['field']
+            if Converter.convert_type(field['group']):
+                f['in'] = Converter.convert_type(field['group'])["type"]
+            f['type'] = Converter.convert_type(field['type'])["type"]
+            f['required'] = not field['optional']
+            f['description'] = field['description']
+            fl.append(f)
+        return fl
+
+    @staticmethod
+    def convert_path(path):
+        """
+        "/accounts/:aid/purchases/:pid" -->  "/accounts/{aid}/purchases/{pid}"
+        """
+        return re.sub(r":(\w*)", r"{\1}", path)
```

Comment:
Should we do '+' here (one or more)? As we should never have an empty /:/ url segment?